### PR TITLE
Replace deprecated use of host.ReportFatalError

### PR DESCRIPTION
**Pull request by exekis**
**Description:**
This update refactors the `Start` method of the `sfxReceiver` to remove the deprecated `host.ReportFatalError` call. The code now uses the new `settings.TelemetrySettings.ReportStatus` method with `component.NewFatalErrorEvent(err)` to report startup errors.

**Link to tracking Issue:**
#30598

**Testing:**
The updated code has been tested to handle error scenarios during the startup of the `sfxReceiver`, and it now properly reports errors using the new telemetry settings.

**Documentation:**
No changes to external documentation were needed as the change is internal to the `sfxReceiver`'s implementation and does not affect the receiver's interface or configuration.

### CODE REVIEW

File: receiver/awsfirehosereceiver/receiver.go [hunk 1]
```diff
@@ -126,13 +126,14 @@ func (fmr *firehoseReceiver) Start(_ context.Context, host component.Host) error
...
 		if errHTTP := fmr.server.Serve(listener); errHTTP != nil && !errors.Is(errHTTP, http.ErrServerClosed) {
-			host.ReportFatalError(errHTTP)
+			fmr.settings.TelemetrySettings.ReportStatus(component.NewFatalErrorEvent(errHTTP))
 		}
 	}()
 
 	return nil
 }
 
+
```
Comment (atoulme) [inline receiver/awsfirehosereceiver/receiver.go:136]:
```suggestion
```

File: receiver/signalfxreceiver/receiver.go [hunk 2]
```diff
@@ -115,7 +115,7 @@ func (r *sfxReceiver) RegisterLogsConsumer(lc consumer.Logs) {
 // Start tells the receiver to start its processing.
 // By convention the consumer of the received data is set when the receiver
 // instance is created.
-func (r *sfxReceiver) Start(_ context.Context, host component.Host) error {
+func (r *sfxReceiver) Start(ctx context.Context, host component.Host) error {
```
Comment (atoulme) [inline receiver/signalfxreceiver/receiver.go:118]: This change is not necessary - try `make lint` to see if this passes.

**Comment by dev-jesse** [comment]
Looks like your changelog check didn't pass. To fix this, you can refer to this portion of the CONTRIBUTING.md file:

> 1. Create an entry file using make chlog-new. This generates a file based on your current branch (e.g. ./.chloggen/my-branch.yaml)
> 2. Fill in all fields in the new file
> 3. Run make chlog-validate to ensure the new file is valid
> 4. Commit and push the file

Or 

>Alternately, copy ./.chloggen/TEMPLATE.yaml, or just create your file from scratch.

**Review by exekis — COMMENTED** [review]
The parameter has been renamed and `make lint` now passes all the tests.

File: receiver/datadogreceiver/receiver.go [hunk 3]
```diff
@@ -47,35 +47,35 @@ func newDataDogReceiver(config *Config, nextConsumer consumer.Traces, params rec
...
+    }
+
+    ddr.address = hln.Addr().String()
+
+    go func() {
+        if err := ddr.server.Serve(hln); err != nil && !errors.Is(err, http.ErrServerClosed) {
+            ddr.params.TelemetrySettings.ReportStatus(component.NewFatalErrorEvent(err))
+        }
+    }()
+    return nil
```
Comment (atoulme) [inline receiver/datadogreceiver/receiver.go:78]: Please revert whitespace changes

File: receiver/datadogreceiver/receiver.go [hunk 4]
```diff
@@ -47,35 +47,35 @@ func newDataDogReceiver(config *Config, nextConsumer consumer.Traces, params rec
 }
 
 func (ddr *datadogReceiver) Start(_ context.Context, host component.Host) error {
```
Comment (atoulme) [inline receiver/datadogreceiver/receiver.go:49]: This file should not be part of this PR

File: .chloggen/fix_deprecated_reportfatalerror.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: signalfxreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fixed an issue in the SignalFx receiver where the context parameter was not utilized properly.
```
Comment (atoulme) [inline .chloggen/fix_deprecated_reportfatalerror.yaml:10]: That's not correct, the changes to signalfxreceiver do not include changing the context parameter.

Given that this change has no impact on user or API changelog, please delete this changelog entry and instead prefix the commit message with `[chore]` or add the label `Skip Changelog` to the PR.

File: receiver/awsfirehosereceiver/receiver.go [hunk 6]
```diff
@@ -105,7 +105,7 @@ var _ http.Handler = (*firehoseReceiver)(nil)
 
 // Start spins up the receiver's HTTP server and makes the receiver start
 // its processing.
-func (fmr *firehoseReceiver) Start(_ context.Context, host component.Host) error {
+func (fmr *firehoseReceiver) Start(ctx context.Context, host component.Host) error {
```
Comment (atoulme) [inline receiver/awsfirehosereceiver/receiver.go:108]: Why is this file changed? I thought the PR was only touching signalfxreceiver.

**Comment by atoulme** [comment]
Closing - let's open separate PRs for each component.
